I1 patron

haynescrossley opened the pull request:
Refactored Patron Class.
Author - Haynes
Mediator - @elliotmitchum 
Reviewer - @amrmosit 

Please review Patron class changes before merging

Thanks,
Haynes

haynescrossley committed:
Update Patron.java

edited variable names to conform with style guidelines

haynescrossley committed:
Update Patron.java

edit constructor variable names to conform with style guidelines

haynescrossley committed:
Update Patron.java

Updated method names to conform with style guidelines

haynescrossley committed:
Update Patron.java

Updated payFine method variable names to conform with style guidelines

haynescrossley committed:
Updated Patron.java

Updated takeOutLoan and dischargeLoan method parameters to conform with styele guidelines. Also added missing brackets from if statements in these methods

haynescrossley committed:
Updated Patron.java

added missing brackets in payFine method

haynescrossley committed:
Update Patron.java

Fix some spacing/formatting

elliotmitchum commented:
@amrmosit  This is ready for your review

haynescrossley committed:
Update Patron,java

added double spacing between methods

haynescrossley committed:
Merge branch 'master' into i1-patron

haynescrossley committed:
Revert "Merge branch 'master' into i1-patron"

This reverts commit cf009c89e2f7d012fd783ee01a44b6c78e0cd2b4, reversing
changes made to 6a8825efed3315996684aa49db9f75180a1a6fcc.

haynescrossley commented:
Changed to draft for now. Will update PR request once Calendar is reviewed and merged into main. Will then update Patron branch with the latest version of Master and re-request PR.

haynescrossley committed:
Merge branch 'master' into i1-patron

haynescrossley committed:
Merge master into i1-patron

Resolve merge issues

haynescrossley commented:
Merged latest version of master into this branch. Includes updated Library, Item, Loan and Calendar classes. I resolved all conflicts from the merge. Hopefully there are no issues.

Ready for review @elliotmitchum

elliotmitchum commented:
@amrmosit This is ready for your review and approval

amrmosit commented on src/library/entities/Patron.java:
```diff
@@ -1,118 +1,116 @@
+
+    
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
```
PR#36 R1-D1-CSG#2: All variable names are to be meaningful

haynescrossley replied:
[Fixed here with this commit](https://github.com/elliotmitchum/ITC205/pull/36/commits/bb18cb3ecee12555f839f8c7e2bd0f7d0e383774)

amrmosit commented on src/library/entities/Patron.java:
```diff
@@ -1,118 +1,116 @@
+          .append(String.format("  Fines Owed :  $%.2f", finesOwing))
+          .append("\n");
+        
+        for (Loan LoAn : currentLoans.values()) {
```
PR#36 R1-D2-CSG#2: All variable names are to start with a lowercase letter and to be in camelBack
hint:LoAn to be loan.

haynescrossley replied:
[Fixed here with this commit](https://github.com/elliotmitchum/ITC205/pull/36/commits/a3ed401a1c8b98aa6e2231bb45909088ecaa6e7e)

amrmosit commented on src/library/entities/Patron.java:
```diff
@@ -1,118 +1,116 @@
+    }
+
+    
+    public double finesOwed() {
```
PR#36 R1-D3-CSG#3: All methods are to be meaningful. Method names should be verbs or verb phrases
hint: finesOwed to be getFinesOwed

haynescrossley replied:
[Fixed with this commit](https://github.com/elliotmitchum/ITC205/pull/36/commits/b0581c296e924a588632c7da322c6d5bd237f584)

amrmosit commented on src/library/entities/Patron.java:
```diff
@@ -1,118 +1,116 @@
+
+    
+    public void takeOutLoan(Loan loan) {
+        if (!currentLoans.containsKey(loan.getId())) {
```
PR#36 R1-D4-CSG#10: The argument list of any metho calls MUST NOT contain any other method calls
hint: add local variable Long loanId = loan.getId();
if(!currentLoans.containsKey(loanId){ // same for line 69

haynescrossley replied:
[Fixed with this commit](https://github.com/elliotmitchum/ITC205/pull/36/commits/20394d2e74ee4cdc1916fef8b02a59b24cb7e928)

amrmosit commented on src/library/entities/Patron.java:
```diff
@@ -1,118 +1,116 @@
+    public void dischargeloan(Loan loan) {
+        if (currentLoans.containsKey(loan.getId())) {
+            currentLoans.remove(loan.getId());
+        }
```
PR#36 R1-D5-CSG#10: The argument list of any metho calls MUST NOT contain any other method calls
same hint as PR#36 R1-D4-CSG#10

haynescrossley replied:
[Fixed here with this commit](https://github.com/elliotmitchum/ITC205/pull/36/commits/3532e98039fc8d9bdc3e56579303ac073dcd90ad)

amrmosit reviewed (changes requested):
There are 5 Defects to be resolved

haynescrossley committed:
Update Patron.java

R1-D1 - refactored StringBuilder variable to be more meaningful

haynescrossley committed:
Update Patron.java

R1-D2 - fixed casing on variable name

haynescrossley committed:
Update Patron.java

R1 - D3 - refactored finesOwed method  to getFinesOwed

haynescrossley committed:
Update Patron.java

RI-D4 - fixed issue with currentLoans method calling other methods in the argument list

haynescrossley committed:
Update Patron.java

R1-D5 - fixed issue with dischargeLoan method calling other methods in the argument list

haynescrossley commented:
@amrmosit I've made changes based on your feedback. Please review.

amrmosit reviewed (approved):
Thanks. All good. 
Branch i1-patron is reviewed and approved. 
@elliotmitchum @haynescrossley

elliotmitchum commented:
Looks good, builds and runs!
